src/Umbraco.Web/Models/Mapping/ContentMapDefinition.cs

```diff
@@ -169,10 +172,24 @@ private IEnumerable<string> GetActions(IContent source)
                 path = source.Path;
             else
             {
-                var parent = _contentService.GetById(source.ParentId);
                 path = parent == null ? "-1" : parent.Path;
             }
 
+            // A bit of a mess, but we need to ensure that all the required values are here AND that they're the right type.
+            if (context.Items.TryGetValue("CurrentUser", out var userObject) &&
+                context.Items.TryGetValue("Path", out var pathObject) &&
+                context.Items.TryGetValue("Permissions", out var permissionsObject) &&
+                userObject is IUser currentUser &&
+                pathObject is string contextPath &&
+                permissionsObject is EntityPermissionSet permissions)
+            {
+                // Both the path and the user is the same, it's safe to assume that the permissions are the same as well.
+                if (contextPath.Equals(path) && umbracoContext.Security.CurrentUser.Id == currentUser.Id)
```

Comment:
Mapping doesn't always occur on a web request and in that case umbracoContext will be null.

Reply:
Yeah, there's also a comment about in the method, at the very top though we check if umbracoContext is null, and if it is we just return `Enumerable.Empty<string>();` so it's handled before we get to that part.